### PR TITLE
add support for subnet

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

File: ray_on_golem/server/services/golem/helpers/demand_config.py [hunk 1]
```diff
@@ -32,28 +30,14 @@ async def get_payload_from_demand_config(self, demand_config: DemandConfigData)
...
         self, demand_config: DemandConfigData, image_url: URL, image_hash: str
     ) -> Payload:
-        @dataclass
-        class CustomRepositoryVmPayload(RepositoryVmPayload):
-            subnet_constraint: str = constraint("golem.node.debug.subnet", "=", default=SUBNET)
-            debit_notes_accept_timeout: int = prop(
-                "golem.com.payment.debit-notes.accept-timeout?", default=240
-            )
-
         params = demand_config.dict(exclude={"image_hash", "image_tag", "outbound_urls"})
```
Comment (lucekdudek) [inline ray_on_golem/server/services/golem/helpers/demand_config.py:33]: Should we exclude`subnet_tag` here too?

File: ray_on_golem/server/services/ray.py [hunk 2]
```diff
@@ -89,6 +90,7 @@ async def create_nodes(
             raise RayServiceError("Node creation is available only after cluster bootstrap!")
 
         logger.info(f"Creating {count} nodes...")
+        logger.debug(f"{self._provider_config=}")
```
Comment (lucekdudek) [inline ray_on_golem/server/services/ray.py:93]: Does it produce a descriptive enough message?

**Review by lucekdudek — APPROVED** [review]
LGTM but please see my comments

File: ray_on_golem/server/services/golem/helpers/demand_config.py (hunk 1 above)
Comment (shadeofblue) [reply ray_on_golem/server/services/golem/helpers/demand_config.py:33]: hmm, indeed... though, it seems to work this way too :D

Comment (shadeofblue) [reply ray_on_golem/server/services/golem/helpers/demand_config.py:33]: maybe...

Comment (shadeofblue) [reply ray_on_golem/server/services/golem/helpers/demand_config.py:33]: but it works all the same anyway :D

Comment (lucekdudek) [reply ray_on_golem/server/services/golem/helpers/demand_config.py:33]: So why add it to the one below?

Comment (shadeofblue) [reply ray_on_golem/server/services/golem/helpers/demand_config.py:33]: that's a good question :)

Comment (shadeofblue) [reply ray_on_golem/server/services/golem/helpers/demand_config.py]: anyway, updated

File: ray_on_golem/provider/node_provider.py [hunk 3]
```diff
@@ -39,19 +39,18 @@ class GolemNodeProvider(NodeProvider):
     def __init__(self, provider_config: dict, cluster_name: str):
         super().__init__(provider_config, cluster_name)
 
-        provider_parameters = provider_config["parameters"]
+        provider_parameters: dict = provider_config["parameters"]
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply ray_on_golem/provider/node_provider.py:42]: @approxit could you eli5 a bit more?

File: ray_on_golem/provider/node_provider.py [hunk 4]
```diff
@@ -39,19 +39,18 @@ class GolemNodeProvider(NodeProvider):
...
-            node_config=NodeConfigData(**(provider_parameters.get("node_config") or {})),
-            ssh_private_key=provider_parameters["_ssh_private_key"],
-            ssh_user=provider_parameters["_ssh_user"],
-        )
+
+        ssh_arg_mapping = {"_ssh_private_key": "ssh_private_key", "_ssh_user": "ssh_user"}
+        provider_parameters = {
+            ssh_arg_mapping.get(k) or k: v for k, v in provider_parameters.items()
+        }
+        self._ray_on_golem_client.create_cluster(provider_parameters)
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply ray_on_golem/provider/node_provider.py:53]: @approxit I updated the interface as well, and when the object is parsed, `node_config` _is_ `NodeConfigData` ... unless I'm missing something... ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (shadeofblue) [reply ray_on_golem/provider/node_provider.py:53]: [ let's discuss it on a call tomorrow morning, thne ]

[the remark this answers is not in the document]
Comment (approxit) [reply ray_on_golem/provider/node_provider.py:53]: Turned out that version of pydantic can recursively parse nested objects - so its fine.
